### ISSUE TITLE
Speed up yaml parsing for pipelines in searchpath

### ISSUE THREAD

**Issue by bryce-turner** [tgen/jetstream#143]
The more pipelines we have in the searchpath, the longer it takes to find the requested version of the pipeline. This is compounded by the length of the yaml as safe_loading will read/parse the entire yaml.

So far the expectation is that the pipeline.yaml starts with something similar to:
```yaml
__pipeline__:
  name: phoenix
  main: main.jst
  description: Human GRCh38 genomics suite
  version: development
```

Instead of reading the entire yaml before deciding that the pipeline doesn't match the version expected, we could read the header only.

This obviously has some caveats, first being that we would need to be strict about the organization of the pipeline.yaml. Additionally would need to decide on how many lines to read.

Solutions to these caveats are that we could fallback to parsing the entire yaml in the event that the quick read does not return a valid pipeline. As for number of lines, reading the first X lines should still match yaml specification (though there might be an issue if there are fewer lines than expected).

**Comment by bryce-turner** [comment]
Instead of forcibly breaking the way yamls are parsed, we can use the LibYAML implementation. Need to check to see if pip install of pyyaml consistently installs this, but it was available in testing without any reinstall needed in my environment.

I also speculated that safe_load might have some overhead, though that doesn't appear to be the case based on the plot below. Default loader is UnsafeLoader. I also attempted to check the BaseLoader as this ignores all additional features of yaml, though it did not like the test set I was providing.  

Example input:
```yaml
- - 0.5488135039273248
  - 0.7151893663724195
  - 0.6027633760716439
- - 0.5448831829968969
  - 0.4236547993389047
  - 0.6458941130666561
- - 0.4375872112626925
  - 0.8917730007820798
  - 0.9636627605010293
 .
 .
 .
 # Total of n*3 lines, or n groups of 3.
```


Plot of performance, we can see that the LibYAML (noted as yaml_c_*) is about 10X faster on average, we are often in the range of about 10<sup>3</sup> to 10<sup>4</sup>, mostly closer to 10<sup>3</sup>. Meaning we should take about 0.1 second to load a yaml vs 1 second or more it's currently taking. For comparison, we also have a plot for json, which is a more simple structure in terms of features than yaml (yaml has some neat [features](https://realpython.com/python-yaml/#compare-loaders-features) depending on the loader). So it makes sense that json loads a lot faster.
![out](https://user-images.githubusercontent.com/40504111/203114324-bf1ecd19-56ce-42fa-9dba-1566c6ea26f6.png)

**Comment by bryce-turner** [comment]
Merged in #151
